resources/assets/js/videos/videoContainer.vue

```diff
@@ -285,6 +306,13 @@ export default {
                     }
                 });
         },
+        getFeatureFromAnnotation(annotation){
+            let points = annotation.points;
+            let xValues = points[0].filter((_,i) => {return i%2===0;});
+            let yValues = points[0].filter((_,i) => {return i%2===1;});
+            let coords = xValues.map((x,i) => [x,yValues[i]]);
+            return new Feature({geometry: new Polygon([coords])});
+        },
```

Comment:
This function should not be necessary if the polygon is checked before it is appended to the "pending annotation" (see comment above). You should be able to use the original Feature directly there.